### PR TITLE
add new example: Python Job Submit/Wait Example

### PR DESCRIPTION
**Pull request by cmoussa1**
This PR looks to add a new example to our workflow-examples repo based on the idea proposed by @garlick in Issue #45. This example contains three parts:

- **submitter_wait_any.py**: job submit/wait for completion in any order

- **submitter_sliding_window.py**: job submit/wait and keep at most a number of jobs active

- **submitter_by_id.py**: job submit/wait for completion in order by jobid

The **submitter** scripts were essentially copy-and-paste from the examples @garlick provided, which can be found here: [waitany](https://github.com/flux-framework/flux-core/blob/master/t/job-manager/submit-waitany.py), [sliding_window](https://github.com/flux-framework/flux-core/blob/master/t/job-manager/submit-sliding-window.py), and [by_id](https://github.com/flux-framework/flux-core/blob/master/t/job-manager/submit-wait.py). The job I used for submission is **compute.py**, a simple Python script used elsewhere in the repo, like [example2](https://github.com/cmoussa1/flux-workflow-examples/tree/example10/example2). 

Since this example already exist in our repo, I didn't include its link in the top-level README. If this gets merged, I'll create a tiny PR that includes its link in `master`.

### CODE REVIEW

**Comment by SteVwonder** [comment]
This is great!  I think this example is great for the ECP tutorial as a follow on example to example 7.

I'll give this a spin and hopefully we can merge it ASAP :)

File: example10/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+### Example 10(a) - Python Job Submit/Wait
```
Comment (dongahn) [inline example10/README.md]: Maybe it is better to split this into two commits:
1. all the python codes
2. this readme

File: example10/submitter_sliding_window.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import flux
+import sys
+import os
+
+from flux import job
+from flux.job import JobspecV1
+
+njobs = 10
+fanout = 2
```
Comment (dongahn) [inline example10/submitter_sliding_window.py]: Variable name, `fanout` isn't really intuitive to me. I have a slight preference to something like `nactives' or `win_size` or `window_size`...

**Comment by cmoussa1** [comment]
Okay, I think I've successfully split up the commits into two - one to add all of the Python files, and one to add **README.md**. Thanks for your suggestions - @dongahn; I went ahead and replaced `fanout` with `window_size`. Force-pushed those changes.

**Comment by dongahn** [comment]
Thanks @cmoussa1. LGTM. 

And I like that you committed the code first and then README! Exactly the sequence I would have taken.

This is good to land from my perspective. @SteVwonder, are you okay if I merge this?

File: example10/compute.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/env python
```
Comment (SteVwonder) [inline example10/compute.py]:
```suggestion
#!/usr/bin/env python
```

File: example10/submitter_by_id.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    njobs = 10
+else:
+    njobs = int(sys.argv[1])
+
+# open connection to broker
+h = flux.Flux()
+
+# create jobspec for compute.py
+compute_jobspec = JobspecV1.from_command(
+    command=["./compute.py", "30"], num_tasks=4, num_nodes=2, cores_per_task=2
```
Comment (SteVwonder) [inline example10/submitter_by_id.py]:
```suggestion
    command=["./compute.py", "10"], num_tasks=4, num_nodes=2, cores_per_task=2
```

For the purposes of interactivity, I think shrinking this window would be a good idea.

File: example10/submitter_sliding_window.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    njobs = int(sys.argv[1])
+if len(sys.argv) == 3:
+    window_size = int(sys.argv[2])
+
+# open connection to broker
+h = flux.Flux()
+
+# create jobspec for compute.py
+compute_jobspec = JobspecV1.from_command(
+    command=["./compute.py", "10"], num_tasks=4, num_nodes=2, cores_per_task=2
```
Comment (SteVwonder) [inline example10/submitter_sliding_window.py]:
```suggestion
    command=["./compute.py", "5"], num_tasks=4, num_nodes=2, cores_per_task=2
```

Same

File: example10/submitter_wait_any.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    njobs = 10
+else:
+    njobs = int(sys.argv[1])
+
+# open connection to broker
+h = flux.Flux()
+
+# create jobspec for compute.py
+compute_jobspec = JobspecV1.from_command(
+    command=["./compute.py", "30"], num_tasks=4, num_nodes=2, cores_per_task=2
```
Comment (SteVwonder) [inline example10/submitter_wait_any.py]:
```suggestion
    command=["./compute.py", "10"], num_tasks=4, num_nodes=2, cores_per_task=2
```

Same

**Comment by cmoussa1** [comment]
Thanks for your suggestions @SteVwonder! I went ahead and edited the `python` path and reduced the running time for `compute.py` jobs in all three of the submitter scripts, rebased, and force-pushed.

**Comment by SteVwonder** [comment]
One other nit, the `by_id` name threw me off for a bit. I was expecting the job ID to play some critical role in the waiting of that script.  I guess technically it does since IDs are loosely ordered, but in the case where the job ID order != submission order, then I think this script name is a misnomer.  Maybe `wait_in_order` is more direct to the actual functionality?

**Comment by SteVwonder** [comment]
Thanks @cmoussa1!

**Comment by cmoussa1** [comment]
> One other nit, the `by_id` name threw me off for a bit. I was expecting the job ID to play some critical role in the waiting of that script. I guess technically it does since IDs are loosely ordered, but in the case where the job ID order != submission order, then I think this script name is a misnomer. Maybe `wait_in_order` is more direct to the actual functionality?

I think `wait_in_order` might make more sense too. I didn't really know what to name it, so I just copied the name from a comment from [the](https://github.com/flux-framework/flux-core/blob/master/t/job-manager/submit-wait.py) test script in `t/job-manager/submit-wait.py`:

`# Submit njobs jobs in a loop, then wait for them by id.` 

I went ahead and renamed the file to `submit_wait_in_order.py`!

File: example10/README.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+1. Allocate three nodes from a resource manager:
+
+`salloc -N3 -ppdebug`
+
+2. Launch a Flux instance on the current allocation by running `flux start` once per node, redirecting log messages to the file `out` in the current directory:
+
+`srun --pty --mpi=none -N3 flux start -o,-S,log-filename=out`
+
+3. Submit the **submitter_byid.py** script, along with the number of jobs you want to run (if no argument is passed, 10 jobs are submitted):
```
Comment (SteVwonder) [inline example10/README.md]:
```suggestion
3. Submit the **submitter_wait_in_order.py** script, along with the number of jobs you want to run (if no argument is passed, 10 jobs are submitted):
```

File: example10/README.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+`./submitter_byid.py 10`
```
Comment (SteVwonder) [inline example10/README.md]:
```suggestion
`./submitter_wait_in_order.py 10`
```

**Comment by cmoussa1** [comment]
Thanks for catching that mistake @SteVwonder! I realized I also forgot some underscores in the other two scripts in **README** as well. Force-pushed those fixes up!

**Comment by cmoussa1** [comment]
Thanks! 🎉

**Pull request merged**